jax/_src/numpy/inspection.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2023 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import jax
+from typing import Tuple
+from jax._src.sharding import Sharding
+from jax._src.lib import xla_client as xc
+from jax._src import dtypes as _dtypes, config
+
+class __array_namespace_info__:
```

Comment:
On second thought, since we're not ready for `jax.numpy` to be an array namespace yet, we should define internal array api utilities like this one in `jax.experimental.array_api` for now.

Other top-level public array API functions in your other PRs should be in `jax.numpy` from the start. Does that sound reasonable?

Reply:
Sure, we can save the internal stuff for last once we're fully compliant.